src/thor/bidirectional_astar.cc

```diff
@@ -934,16 +936,47 @@ std::vector<std::vector<PathInfo>> BidirectionalAStar::FormPath(GraphReader& gra
   // Special case code if the last edge of the forward path is
   // the destination edge - update the elapsed time
   if (edgelabels_reverse_[idx2].predecessor() == kInvalidLabel) {
-    // destination is on a different edge than origin
+    // destination is on a different edge than origin because of that we know that the forward path
+    // did not care about partial distance along the edge when it found the edge so what we need to do
+    // is look at the second to last edge and use the reverse path (who does care about partial
+    // distance on the destination edge) to recompute the elapsed time and cost
     if (path.size() > 1) {
-      path.back().elapsed_time =
-          path[path.size() - 2].elapsed_time + edgelabels_reverse_[idx2].cost().secs;
-      path.back().elapsed_cost =
-          path[path.size() - 2].elapsed_cost + edgelabels_reverse_[idx2].cost().cost;
+      path.back().elapsed_time = path[path.size() - 2].elapsed_time +
+                                 edgelabels_reverse_[idx2].cost().secs +
+                                 edgelabels_forward_[idx1].transition_secs();
+      path.back().elapsed_cost = path[path.size() - 2].elapsed_cost +
+                                 edgelabels_reverse_[idx2].cost().cost +
+                                 edgelabels_forward_[idx1].transition_cost();
```

Comment:
this block was the first bug. when the route it short enough, it can happen that the reverse path is only one edge, the last edge that the forward path found. in this case we were taking the elapsed time from the second to last edge and using it to recompute the elapsed time taking into acount the trimming of the last edge by the snap location of the destination. what we werent doing was actually remembering that the elapsed time/cost also contains the transition (turn) cost. so we've added that back here from the forward path where it is stored.

we noticed this by looking at map matching tests that compare the map match against the router. the map matcher had a longer time because its form path method properly included the turn cost. its a lot less complicated because its not bidirectional ;)